src/backends/perf_pt/decode.c

```diff
@@ -153,8 +154,11 @@ perf_pt_init_block_decoder(void *buf, uint64_t len, int vdso_fd, char *vdso_file
         goto clean;
     }
 
+    // Use image cache to speed up decoding.
+    struct pt_image_section_cache *iscache = pt_iscache_alloc(NULL);
```

Comment:
I think this value needs to be checked to see if it's `NULL`.

Reply:
Agreed:

> Returns a new traced memory image section cache on success, NULL otherwise.

Reply:
Fixed in [75616b9](https://github.com/ykjit/hwtracer/pull/98/commits/75616b925502e6fa1bc48e8de6b864086d3d1626)